addons/web/static/src/views/calendar/calendar_model.js

```diff
@@ -662,7 +670,7 @@ export class CalendarModel extends Model {
                     filters.push({
                         id: value,
                         [fieldName]: rawValue,
-                        colorIndex: record.colorIndex,
+                        ...this.addFilterFields(record, filterInfo),
```

Comment:
Cannot we change the `colorIndex` in record to have the one we want? 🤔 

I don't know if it is possible. 🙂

Reply:
Mmh we still need the method because we also need the resource_type of the record ( so we know if we have to display a picture or a colored icon in the filterPanel). As for the colorIndex, according the the type of resource we don't fetch the same field to fill the colorIndex

Reply:
Indeed. 🙂